src/main/java/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package service.impl;
+
+import dto.user.UserRegistrationRequestDto;
+import dto.user.UserResponseDto;
+import exceptions.RegistrationException;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import mapper.UserMapper;
+import model.Role;
+import model.User;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+import repository.RoleRepository;
+import repository.UserRepository;
+import service.UserService;
+
+@Service
+@RequiredArgsConstructor
+@Component
```

Comment:
```suggestion
@Service
@RequiredArgsConstructor
```
One annotation enough )